Lib/sre_parse.py

```diff
@@ -432,11 +432,9 @@ def _escape(source, escape, state):
 def _uniq(items):
     if len(set(items)) == len(items):
         return items
-    newitems = []
-    for item in items:
-        if item not in newitems:
-            newitems.append(item)
-    return newitems
+    seen_items = set()
+    seen_items_add = seen_items.add
+    return [item for item in items if not (item in seen_items or seen_items_add(item))]
```

Comment:
How about this?

```py
def _uniq(items):
    return list(dict.fromkeys(items))
```

Reply:
Since `dict.fromkeys()` already jumbled up the item unless there's a need to return a list, returning a set might be more optimal when checking for unique, i.e. 

```
def _uniq(items):
    return set(dict.fromkeys(items))
```

Reply:
`set(dict.fromkeys(X))` is totally useless.  When you need `set`, `set(X)` is enoguh.
But in this case, we need list, not set.  So I used `list(dict.fromkeys(X))`.

Reply:
We indeed need a list. And I guess that preserving the order is important, looking at the initial implementation.

I'm fine with `list(dict.fromkeys(items))` (and it's much prettier), but I have 2 minor concerns:
- `list(dict.fromkeys(items))` *might* be a bit slower (though I didn't check, and the difference is negligeible I think)
- we have to make sure this won't be backported to python < 3.7 as the dict order is not necessarily the insert order for python < 3.7

What do you think?

Reply:
> * `list(dict.fromkeys(items))` _might_ be a bit slower (though I didn't check, and the difference is negligeible I think)

I believe `list(dict.fromkeys(items))` is significantly faster than `[it for it in items if it not in seen or seen.add(it)]`

> * we have to make sure this won't be backported to python < 3.7 as the dict order is not necessarily the insert order for python < 3.7 (but again, the bug only affects >=3.7 and the function does not exist on <3.7, so it won't be backported anyway)

You said "sre_parse._uniq function, which does not exist in Python <= 3.6."
So no need to think about 3.6.

Reply:
@methane you were right, list(dict.fromkeys(items)) seems to be a bit faster. Changed ✅.